cpp/src/arrow/util/compression.h

```diff
@@ -174,5 +174,55 @@ class ARROW_EXPORT Codec {
   virtual Status Init();
 };
 
+// ----------------------------------------------------------------------
+// Codec factories
+
+namespace internal {
```

Comment:
These could also be moved to "compression_internal.h" if they are seen as a nuisance here

Reply:
That would be nice IMHO.